setupext.py

```diff
@@ -731,8 +731,8 @@ def get_extensions(self):
             'matplotlib.backends._macosx', [
                 'src/_macosx.m'
             ])
-        ext.extra_compile_args.extend(['-Werror'])
-        ext.extra_link_args.extend(['-framework', 'Cocoa'])
+        ext.extra_compile_args.extend(['-Werror', '-mmacosx-version-min=10.12'])
+        ext.extra_link_args.extend(['-framework', 'Cocoa', '-mmacosx-version-min=10.12'])
```

Comment:
Do we actually need this? I think the environment variable `MACOSX_DEPLOYMENT_TARGET` should be used instead, which you already have defined above. This says they are synonymous, and more importantly that this flag is actually a gcc flag that clang supports for now, but potentially not into the future.
https://stackoverflow.com/questions/25352389/what-is-the-difference-between-macosx-deployment-target-and-mmacosx-version-min

Reply:
I've removed this change from my latest commit to see if it's needed.

Reply:
It turns out we didn't need this - builds seem to be fine with just the environment variable set.